src/test/scala/scala/tools/refactoring/tests/implementations/imports/OrganizeImportsTest.scala

```diff
@@ -4410,4 +4411,902 @@ class OrganizeImportsTest extends OrganizeImportsBaseTest {
+    """
+    package acme.duration
+
+	  trait DurationConvensions {
+      def baz = "baz"
+	  }
+
+	  object Duration {
+      def foo = "Duration"
+	  }
+    """ isNotModified
+
+    """
+    package acme
+
+    package object duration {
+      implicit class bar(b: String) extends DurationConvensions
+    }
+    """ isNotModified
+
+    """
+    package acme
+
+    class Promise[T]
+    """ isNotModified
+
+    """
+    /*<-*/
+    package test
+    import acme.Promise
+    import acme.duration.Duration
+
+    object A {
+      def f = {
+        import acme.duration._
+        val p = new Promise[Int]
+        println(p)
+        "5".baz
+      }
+      def g = Duration.foo
+    }
+    """ isNotModified
+  } applyRefactoring organizeCustomized(dependencies = Dependencies.RecomputeAndModify)
+
+  @Test
+  def shouldNotRemoveImportFromSamePackageWhenTypeUsedInImplicit_relativePath() = new FileSet {
+    """
+    package api
+
+    trait TypeTags {
+      trait TypeTag[T]
+      def typeOf[T](implicit tt: TypeTag[T]) = ???
+    """ isNotModified
+
+    """
+    package tested
+
+    object inner {
+      type Inner[T, U] = T with U
+    }
+    """ isNotModified
+
+    """
+    /*<-*/
+    package tested
+
+    import api.TypeTags
+
+    class Test extends TypeTags {
+      import inner.Inner
+
+      implicit val innerTT = new TypeTag[Inner[_, _]] {}
+
+      val test = typeOf[Inner[_, _]]
+    }
+    """ isNotModified
+  } applyRefactoring organizeCustomized(dependencies = Dependencies.RecomputeAndModify)
+
+  @Test
+  def shouldNotRemoveImportFromSamePackageWhenTypeUsedInImplicit_absolutePath() = new FileSet {
+    """
+    package api
+
+    trait TypeTags {
+      trait TypeTag[T]
+      def typeOf[T](implicit tt: TypeTag[T]) = ???
+    """ isNotModified
+
+    """
+    package tested
+
+    object inner {
+      type Inner[T, U] = T with U
+    }
+    """ isNotModified
+
+    """
+    /*<-*/
+    package tested
+
+    import api.TypeTags
+
+    class Test extends TypeTags {
+      import tested.inner.Inner
+
+      implicit val innerTT = new TypeTag[Inner[_, _]] {}
+
+      val test = typeOf[Inner[_, _]]
+    }
+    """ isNotModified
+  } applyRefactoring organizeCustomized(dependencies = Dependencies.RecomputeAndModify)
+
+  @Test
+  def shouldNotRemoveImportFromSamePackageWhenInnerTraitUsed() = new FileSet {
+    """
+    package tested
+
+    object inner {
+      type Inner = Int
+    }
+    """ isNotModified
+
+    """
+    /*<-*/
+    package tested
+
+    import inner.Inner
+
+    class Test {
+      val test: Inner = ???
+    }
+    """ isNotModified
+  } applyRefactoring organizeCustomized(dependencies = Dependencies.RecomputeAndModify)
+
+  @Ignore("FIXME")
+  @Test
+  def shouldNotRemoveImportWhenTypeParametrizedByObjectInConstructor() = new FileSet {
+    """
+    package org.acme
+
+    trait Inner {
+      type Inner
+    }
+
+	  case class Toolbox[T <: Inner](inner: T)
```

Comment:
no, it is another failing scenario, sorry for confusion, should go with separate commit